src/Services/Implementations/DefaultAlgoliaIndexService.cs

```diff
@@ -26,9 +26,22 @@ public DefaultAlgoliaIndexService(ISearchClient searchClient)
         }
 
 
+        public string GetIndexName(string indexName)
+        {
+            return indexName;
+        }
+
+
         public ISearchIndex InitializeIndex(string indexName)
         {
             return searchClient.InitIndex(indexName);
         }
+
+
+        public void SetIndexSettings(string indexName, IndexSettings indexSettings)
```

Comment:
This could be a little confusing because in `public ISearchIndex InitializeIndex(string indexName)` above, the expectation is that the `indexName` is already customized by using `public string GetIndexName(string indexName)`, but here the assumption is the `string indexName` parameter has not yet been customized.

The problem isn't so much the various customization points as its using `string` (a primitive) to represent 2 different types of data - customized index names and uncustomized names.

What about using a custom type like:

```csharp
public struct IndexName
{
	public IndexName(string value)
	{
		Value = value;
	}
	
	public string Value { get; set; } = "";
}
```

`string GetIndexName(string indexName)` could be modified to return a `IndexName` instance instead.

Any method expecting an unmodified `IndexName` would take a `string` parameter and any method expecting a potentially modified value would take an `IndexName` parameter.

Reply:
ha, I didn't notice you commented on the PR too :) 

Is there a case where unmodified index is needed?

Reply:
I don't believe it is needed anywhere.

I had hoped, before this integration was released, that index names would come from configuration (since there's nothing worth managing in the CMS/database).

Having the index name be an attribute property means it has to be defined at compile time which is very limiting. It's a similar limitation to how Azure Search indexes get their name from the value specified in the CMS upon creation - this makes having different index names per environment much more complex.

Reply:
`SetIndexSettings` was removed in https://github.com/Kentico/xperience-algolia/pull/24/commits/888a501ff25850ecc57517fdde90dfed3deed90d as it was unnecessary (settings can be customized via [GetIndexSettings](https://github.com/Kentico/xperience-algolia/blob/master/src/Services/Implementations/DefaultAlgoliaRegistrationService.cs#L79)).